artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/codec/PageCountPendingImpl.java

```diff
@@ -43,6 +44,8 @@ public PageCountPendingImpl(long queueID, long pageID, int inc) {
 
    long pageID;
 
+   long size;
```

Comment:
Should default be -1

Reply:
Why would we use -1 and not 0? These values default to 0 because we don't want the counter to change if the size is not set.  If we default to -1 then the counters will go negative where as if they are 0 when not set then nothing changes with the counters which is what we want.

Reply:
There’s a difference between no size (0) and not being set (-1) the logic calculating should be aware of this as -1 means it would need to be calculated

Reply:
Eg it’s typical for -1 to be used to represent null vs 0 as no size , slight semantic difference but important none the less

Reply:
The issue is backwards compatibility.  These values are now being stored to the journal and old records won't have a size.  So by defaulting to a 0 value for when an old record is it makes everything simple.  Yes I could make it -1 but then I have to add checks for -1 versus just using 0 as a default and then the logic is clean as we don't need to handle negative edge cases.

Reply:
@michaelandrepearce - did you have in mind having the getter in the class handle the -1 (ie have the getSize() method check for -1 and return 0 if negative) or having the getSize() method return the -1 as is and then have the code using the value explicitly do the check for negative?